src/Compilers/CSharp/Portable/Symbols/Source/SourceMethodSymbolWithAttributes.cs

```diff
@@ -395,55 +395,65 @@ internal sealed override ObsoleteAttributeData ObsoleteAttributeData
         }
 
 #nullable enable
-        internal sealed override UnmanagedCallersOnlyAttributeData? UnmanagedCallersOnlyAttributeData
+        internal sealed override UnmanagedCallersOnlyAttributeData? GetUnmanagedCallersOnlyAttributeData(bool forceComplete)
         {
-            get
+            if (syntaxReferenceOpt is null)
             {
-                if (syntaxReferenceOpt is null)
-                {
-                    // no references -> no attributes
-                    return null;
-                }
+                // no references -> no attributes
+                return null;
+            }
 
-                var lazyCustomAttributesBag = _lazyCustomAttributesBag;
-                if (lazyCustomAttributesBag is null)
-                {
-                    return UnmanagedCallersOnlyAttributeData.Uninitialized;
-                }
+            if (forceComplete)
+            {
+                _ = this.GetAttributes();
+            }
 
-                if (lazyCustomAttributesBag.IsDecodedWellKnownAttributeDataComputed)
-                {
-                    var data = (MethodWellKnownAttributeData?)lazyCustomAttributesBag.DecodedWellKnownAttributeData;
-#if DEBUG // Can remove ifdefs and replace with Conditional after https://github.com/dotnet/roslyn/issues/47463 is fixed
-                    verifyDataConsistent((CommonMethodEarlyWellKnownAttributeData?)lazyCustomAttributesBag.EarlyDecodedWellKnownAttributeData, data);
+            var lazyCustomAttributesBag = _lazyCustomAttributesBag;
+            if (lazyCustomAttributesBag is null || !lazyCustomAttributesBag.IsEarlyDecodedWellKnownAttributeDataComputed)
```

Comment:
>!lazyCustomAttributesBag.IsEarlyDecodedWellKnownAttributeDataComputed [](start = 51, length = 69)

Is it possible that IsDecodedWellKnownAttributeDataComputed returns true, but IsEarlyDecodedWellKnownAttributeDataComputed returns false? If not, an assert somewhere (perhaps in IsEarlyDecodedWellKnownAttributeDataComputed) could be helpful.

Reply:
I don't believe so, because the code that sets the normal data is the same path that sets the early data, it just bails after setting the early data if early is all that's requested: [http://sourceroslyn.io/#Microsoft.CodeAnalysis.CSharp/Symbols/Symbol_Attributes.cs,262](http://sourceroslyn.io/#Microsoft.CodeAnalysis.CSharp/Symbols/Symbol_Attributes.cs,262). I can add an assert to the two getters though.

---
In reply to: [491021696](https://github.com/dotnet/roslyn/pull/47651#discussion_r491021696) [](ancestors = 491021696)